src/c-writer.cc

```diff
@@ -212,6 +213,8 @@ class CWriter {
   std::string DefineLocalScopeName(std::string_view);
   std::string DefineStackVarName(Index, Type, std::string_view);
 
+  static void SerializeFuncType(const FuncType&, std::string&);
```

Comment:
If we can move this function into its own .cc file that depends only on wasm-rt.h and sha256.h, that would be really helpful for the rlbox use case. We could probably move this functionality into the current `sha256.c` file and rename that to something like `func-type-serialize.c`.

Reply:
Ok cool, let's iterate on this and I think we can find something that works for everybody. The most challenging part is going to be about mangling the individual types (e.g. how to turn an i32 into "i") in a way that's convenient. Right now, that means including wabt/type.h, which brings in wabt/base-types.h, wabt/string-format.h, and (this last one is probably optional) wabt/config.h. So, a small number of headers, but not as small as you asked for. (Unfortunately I don't think wasm2c/libwabt can depend on wasm-rt.h -- that's crossing a boundary between transpiler and runtime.)

Q: I pushed a version of this (with documentation in the wasm2c/README.md) -- is it acceptable to you? If so, awesome. :smile: 

Q: If not, would you prefer a more explicit serialization that doesn't depend on anything in libwabt, and that you could feasibly implement outside? We could make the input to the hash function be something like "(param i32 f32 i32) (result i64)" (or whatever), and then you'd hash that string with whatever SHA-256 implementation you prefer. Would that be preferable?

[Q: A third option (but more work for you and I suspect not one that makes you happy) would be to stop passing funcrefs between the host and a wasm2c-generated module. Instead you'd have to make a shim module that imports the host function (directly), and imports the library's table, and fills in the funcref entry in the table. Then you'd run that shim module through wasm2c, and then just use it at runtime. That way all the functypes would be generated by wasm2c. But I don't want you to have to change your whole application here; I think we can find a way for you to generate functypes at runtime that isn't a major burden.]